pdudaemon/drivers/cleware.py

```diff
@@ -42,9 +44,29 @@ def __init__(self, hostname, settings):
         self.settings = settings
         self.serial = settings.get("serial", u"")
         log.debug("serial: %s" % self.serial)
-
         super().__init__()
 
+    def switch4_serial(self, dev_dict):
+        dev = hid.device()
+        dev.open_path(dev_dict["path"])
+        serial = 0
+        for i in range(8, 15):
+            b = int(chr(self.read_byte(dev, i)), 16)
+            serial *= 16
+            serial += b
+        dev.close()
+        return serial
+
+    def device_path(self):
+        for dev_dict in hid.enumerate(CLEWARE_VID, CLEWARE_SWITCH1_PID):
+            if int(dev_dict["serial_number"], 16) != CLEWARE_SWITCH4_SERIAL:
+                continue
+            if self.serial == self.switch4_serial(dev_dict):
```

Comment:
Potentially worth adding a debug print for the actual serial numbers found (either always or simply listing them if the configured serial didn't match any)

Reply:
I added a debug print just before line 64, to debug print the serial with value `int(dev_dict["serial_number"], 16)`